### ISSUE TITLE
Enhance MD040 to validate against a list of supported languages (one of which may be no language)

### ISSUE THREAD

**Issue by DavidAnson** [markdownlint/markdownlint#87]
As part of bringing a set of Markdown files into compliance, I got feedback from the project maintainer that they're not picky about having a language everywhere - but if there IS a language, they'd like to be sure it's a valid one. I'm thinking this could be done via an optional `languages` parameter that specified an array of language names. If it's clear to allow an empty name, that may suffice for the "no language" option - otherwise maybe a magic name ("[NONE]"?) or else a second optional parameter `allow_none`. However, what's weird about the latter is that turning on MD040 and `allow_none` is a complete no-op - which is why it'd be nice to make that part of the "supported language list".

**Comment by kachkaev** [comment]
👍

It’d be also cool to configure language aliases for improved feedback & autofixes. E.g. ` ```bash ` and ` ```sh ` can be automatically replaced with ` ```shell ` based on that.
